pkg/goDB/engine/query.go

```diff
@@ -392,23 +408,62 @@ func createWorkManager(dbPath string, iface string, tfirst, tlast int64, query *
 	return
 }
 
-func parseIfaceList(dbPath string, ifaceList string) ([]string, error) {
+func parseIfaceListWithCommaSeparatedString(lister types.InterfaceLister, dbPath string, ifaceList string) ([]string, error) {
 	if ifaceList == "" {
 		return nil, errors.New("no interface(s) specified")
 	}
 
-	if types.IsAnySelector(ifaceList) {
-		ifaces, err := info.GetInterfaces(dbPath)
-		if err != nil {
-			return nil, err
+	allIfaces, err := lister.GetInterfaces(dbPath)
+	if err != nil {
+		return nil, err
+	}
+
+	selectedValidIfaces, negationFilters, validationErr := types.ValidateAndSeparateFilters(ifaceList)
+	if validationErr != nil {
+		return nil, validationErr
+	}
+
+	// add interfaces
+	var result []string
+	for _, iface := range selectedValidIfaces {
+		if types.IsAnySelector(iface) {
+			result = allIfaces
+		} else if slices.Contains(allIfaces, iface) {
+			result = append(result, iface)
+		}
+
+	}
+
+	// remove interfaces
+	for _, notIface := range negationFilters {
+		if slices.Contains(result, notIface) {
+			for i, v := range result {
+				if v == notIface {
+					// Remove the element by slicing
+					result = append(result[:i], result[i+1:]...)
+				}
+			}
 		}
-		return ifaces, nil
 	}
+	return result, nil
+}
 
-	ifaces, err := types.ValidateIfaceNames(ifaceList)
+func parseIfaceListWithRegex(lister types.InterfaceLister, dbPath string, ifaceRegexp string) ([]string, error) {
```

Comment:
Based on the change of the `InterfaceLister` interface, you can now also omit the `dbPath` parameter of this function.